pkg/trace/init.go

```diff
@@ -0,0 +1,60 @@
+package trace
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"log"
+	"sync"
+)
+
+var (
+	once     sync.Once
+	exporter *otlptrace.Exporter
+)
+
+func initExporter() {
+	var err error
+	exporter, err = otlptrace.New(
+		context.Background(),
+		otlptracegrpc.NewClient(
+			otlptracegrpc.WithInsecure(),
+			otlptracegrpc.WithEndpoint("localhost:4317"),
+		),
+	)
+	if err != nil {
+		log.Fatal("Could not set exporter: ", err)
+	}
+}
+
+func InitTracerAuto() func(context.Context) error {
+	once.Do(initExporter)
+
+	resources, err := resource.New(
+		context.Background(),
+		resource.WithAttributes(
+			attribute.String("service.name", "tangseng-service"),
+			attribute.String("application", "tangseng-app"),
+		),
+	)
+	if err != nil {
+		log.Fatal("Could not set resources: ", err)
+	}
+
+	otel.SetTracerProvider(
+		sdktrace.NewTracerProvider(
+			sdktrace.WithSampler(sdktrace.AlwaysSample()),
+			sdktrace.WithSpanProcessor(sdktrace.NewBatchSpanProcessor(exporter)),
+			sdktrace.WithSyncer(exporter),
+			sdktrace.WithResource(resources),
+		),
+	)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+
+	return exporter.Shutdown
```

Comment:
Have you tested trace ? I think this should be appeared in the response body. 
such as :
```json
{
    "code": 200,
    "status": 200,
    "msg": "success",
    "errors": null,
    "data": {},
    "timestamp": 1682042879,
    "trace_id": "0a98685f6441efffb72d687be99381aa"
}
```